_posts/2015-10-19-introduction_to_clojurescript.md

```diff
@@ -0,0 +1,269 @@
+
+* __Performance__
+
+    ClojureScript utilizes [Google's Closure Compiler](http://code.google.com/closure/compiler/docs/api-tutorial3.html). This allows ClojureScript to utilize dead code elimination and other features.
+
+* __Live Coding__
+
+    The ClojureScript ecosystem provides many tools to enable 'live coding'. This means that once you change your code, it's instantly reflected in your live project. In this article, we'll be looking at [Figwheel](https://github.com/bhauman/lein-figwheel).
+
+* __Code Reuse__
+
+    ClojureScript can be run universally which is also called "isomorphically." This means you can run the same code on your client and your server. This has become a popular pattern in the Node.js ecosystem. In addition, ClojureScript can import libraries from the Node and Java ecosystems.
+
+## Setting up the Clojure(Script) tool chain
+In this article, we will be installing the tool chain on a Mac OSX Environment. We will need the latest Java version.
+
+### Installing Leiningen
+Leiningen is a build tool for Clojure projects. It will 
+http://leiningen.org/#install
+
+`brew install leiningen`
+
+### Using the Repl
+Now that we have Leningen installed, we can start getting familiar with ClojureScript syntax.
+
+Execute the following command at a terminal: `lein repl`
+
+You should get output that looks similar to:
+
+```
+$ lein repl
+nREPL server started on port 58371 on host 127.0.0.1 - nrepl://127.0.0.1:58371
+REPL-y 0.3.7, nREPL 0.2.10
+Clojure 1.7.0
+Java HotSpot(TM) 64-Bit Server VM 1.6.0_65-b14-466.1-11M4716
+    Docs: (doc function-name-here)
+          (find-doc "part-of-name-here")
+  Source: (source function-name-here)
+ Javadoc: (javadoc java-object-or-class-here)
+    Exit: Control+D or (exit) or (quit)
+ Results: Stored in vars *1, *2, *3, an exception in *e
+
+user=> 
+```
+
+We're now in a ClojureScript Repl. This will allow us quickly execute ClojureScript and view the result.
+
+
+## ClojureScript Syntax
+ClojureScript is a functional language. This means it has functions and limited additional language constructs. It is a Lisp style language.
+
+### Primitives
+
+* Number
+
+    ```
+    user=> 1.23
+    1.23
+    ```
+
+
+* String
+
+    ```
+    user=> "foo"
+    "foo"
+    ```
+
+
+* Vector (array)
+    
+    ```
+    user=> [:bar 3.14 "hello"]
+    [:bar 3.14 "hello"]
+    ```
+
+
+* Map (associative arrays)
+
+    ```
+    user=> {:msg "hello" :pi 3.14 :primes [2 3 5 7 11 13]}
+    {:msg "hello", :pi 3.14, :primes [2 3 5 7 11 13]}
+    ```
+
+* Keyword (used to access Maps)
+
+    ```
+    user=> :foo
+    :foo
+    ```
+
+
+* Set (distinct array)
+    
+    ```
+    user=> #{:bar 3.14 "hello"}
+    #{"hello" 3.14 :bar}
+    ```
+
+### Functions all the way down
+Functions are the building blocks of ClojureScript. To define a function, you even call a function `defn`.
```

Comment:
this is confusing and tripped me up: "you even call a function `defn`" - I thought it meant I had to name my function `defn`. Maybe something like "You even define your own functions using the built in `defn` function"